gammapy/modeling/tests/test_parameter.py

```diff
@@ -186,3 +184,16 @@ def test_parameters_autoscale():
     pars.autoscale()
     assert_allclose(pars[0].factor, 2)
     assert_allclose(pars[0].scale, 10)
+
+
+def test_subcovar():
+    pars_0 = Parameters([Parameter("a", 10), Parameter("b", 20), Parameter("c", 30)])
+    pars_1 = Parameters(list(pars_0.unique_parameters)[:2])
+    pars_0.covariance = np.array([[3, 4, 5], [7, 8, 9], [10, 11, 12]])
+
+    pars_1.set_subcovariance(pars_0)
+    assert np.all(pars_1.covariance == np.array([[3, 4], [7, 8]]))
```

Comment:
Never call `assert np.all(a==b)`
It will not print the two arrays if there's a different in the test report.
Always call `assert_allclose` or `assert_equal` from `numpy.testing` instead.